src/Nfield.Quota/QuotaLevelDefinition.cs

```diff
@@ -2,9 +2,39 @@
 
 namespace Nfield.Quota
 {
-    public class QuotaLevelDefinition
+    public class QuotaLevelDefinition : IEquatable<QuotaLevelDefinition>
     {
         public Guid Id { get; set; }
         public string Name { get; set; }
+
+        public static bool operator ==(QuotaLevelDefinition left, QuotaLevelDefinition right)
+        {
+            return left?.Equals(right) ?? false;
+        }
+
+        public static bool operator !=(QuotaLevelDefinition left, QuotaLevelDefinition right)
+        {
+            return !(left == right);
+        }
+
+        public override bool Equals(object obj)
+        {
+            var other = obj as QuotaLevelDefinition;
+            return other != null && Equals(other);
```

Comment:
`other != null` always returns `true`, so remove it or use `Object.ReferenceEquals(other, null)` instead